messaging/src/test/java/org/axonframework/utils/AsyncInMemoryStreamableEventSource.java

```diff
@@ -123,18 +123,18 @@ public MessageStream<EventMessage> open(@Nonnull StreamingCondition condition, @
 
     @Override
     public CompletableFuture<TrackingToken> latestToken(@Nullable ProcessingContext context) {
-        return CompletableFuture.completedFuture(null);
-    }
-
-    @Override
-    public CompletableFuture<TrackingToken> firstToken(@Nullable ProcessingContext context) {
         return CompletableFuture.completedFuture(
                 eventStorage.isEmpty()
                         ? null
                         : new GlobalSequenceTrackingToken(eventStorage.lastKey() + 1)
         );
     }
 
+    @Override
+    public CompletableFuture<TrackingToken> firstToken(@Nullable ProcessingContext context) {
+        return CompletableFuture.completedFuture(new GlobalSequenceTrackingToken(-1));
+    }
+
```

Comment:
There was a mistake in the implementation. Now it's easier with the new names to do not mix it :)